### ISSUE TITLE
regression: low FPS on https://servo.org/bhtml-newtab/webrender-demos/moire.html

### ISSUE THREAD

**Issue by Darkspirit** [servo/webrender#2706]
URL: https://servo.org/bhtml-newtab/webrender-demos/moire.html
Bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1457683

> There are multiple regressions.
> 2018-02-01 ~60 fps
> 2018-03-01 ~45 fps
> 2018-04-27 ~28 fps

https://bugzilla.mozilla.org/show_bug.cgi?id=1457683#c6 summarized:
> First regression:
> Press F11. good = ~60 fps. bad = 34-56 fps
> 2018-02-20 https://github.com/servo/webrender/compare/4af31b8aa79d5a1f3c0242dea6be4876c71075c5...e8d2ffb404a85651fe08a6d09abbece9bd2b9182

> Second regression:
> Press F11. good = 34-56 fps. bad = *unpleasant coil whine* and 25-31 fps.
> 2018-04-09 https://github.com/servo/webrender/compare/a05186dc536d68efafcf832f38303162bf552897...df73569c5a9fba35a7c268d176a3eb0d6c52b249

**Comment by glennw** [comment]
Yep, those sound good. I think the biggest win (which @pcwalton mentioned to me) is that we should tessellate the borders we are drawing when they have very big radii, to reduce overdraw. Once I have ported borders to brushes, I *think* it's fairly straightforward to use the segment infrastructure to break up very large border radii rects into small, tesselated segments.

**Comment by kaitk** [comment]
Didn't want to add a new issue, so just commenting on this for now.

On Windows 10, Nvidia GTX 1070 and latest drivers (416.34): 

1. Moire demo is still seriously stuttering on color transitions (otherwise it's almost fluid).
https://servo.org/bhtml-newtab/webrender-demos/moire.html
2. The transparent-rects demo shows corruptions while initially loading it ([image](https://imgur.com/a/Hhcy5hI)). It's strange that sometimes after refreshing, or keeping it open for some time, it displays them correctly:
https://servo.org/bhtml-newtab/webrender-demos/transparent_rects.html
3. Hard to explain, but the spheres demo is sometimes "jiggling"  (left and right) on the X axis, instead of smoothly rotating counter-clockwise
https://servo.org/bhtml-newtab/webrender-demos/spheres.html

On macOS Mojave, Macbook Pro 15" with Radeon 560X I'm not able to replicate the second problem, but the 1. and 3. remain (and performance is considerably worse)

**Comment by kvark** [comment]
Looking at the first case again - our behavior has changed (no clip masks, just cached images now), but in general we still suffer from the same thing: too much pixels to process. And the very same optimization (for symmetry) could apply.
There is probably other things we can do to help this, but I wouldn't call it high priority - the case is super artificial.
